crates/nu-cmd-lang/src/core_commands/version.rs

```diff
@@ -77,68 +84,36 @@ pub fn version(engine_state: &EngineState, call: &Call) -> Result<PipelineData,
     // - installed_plugins
     let mut record = Record::with_capacity(17);
 
-    record.push(
-        "version",
-        Value::string(env!("CARGO_PKG_VERSION"), call.head),
-    );
-
-    push_version_numbers(&mut record, call.head);
-
-    let version_pre = Some(build::PKG_VERSION_PRE).filter(|x| !x.is_empty());
-    if let Some(version_pre) = version_pre {
-        record.push("pre", Value::string(version_pre, call.head));
-    }
-
-    record.push("branch", Value::string(build::BRANCH, call.head));
-
-    let commit_hash = option_env!("NU_COMMIT_HASH");
-    if let Some(commit_hash) = commit_hash {
-        record.push("commit_hash", Value::string(commit_hash, call.head));
-    }
-
-    let build_os = Some(build::BUILD_OS).filter(|x| !x.is_empty());
-    if let Some(build_os) = build_os {
-        record.push("build_os", Value::string(build_os, call.head));
-    }
-
-    let build_target = Some(build::BUILD_TARGET).filter(|x| !x.is_empty());
-    if let Some(build_target) = build_target {
-        record.push("build_target", Value::string(build_target, call.head));
-    }
+    record.push("version", Value::string(env!("CARGO_PKG_VERSION"), span));
 
-    let rust_version = Some(build::RUST_VERSION).filter(|x| !x.is_empty());
-    if let Some(rust_version) = rust_version {
-        record.push("rust_version", Value::string(rust_version, call.head));
-    }
+    push_version_numbers(&mut record, span);
 
-    let rust_channel = Some(build::RUST_CHANNEL).filter(|x| !x.is_empty());
-    if let Some(rust_channel) = rust_channel {
-        record.push("rust_channel", Value::string(rust_channel, call.head));
-    }
+    push_non_empty(&mut record, "pre", build::PKG_VERSION_PRE, span);
 
-    let cargo_version = Some(build::CARGO_VERSION).filter(|x| !x.is_empty());
-    if let Some(cargo_version) = cargo_version {
-        record.push("cargo_version", Value::string(cargo_version, call.head));
-    }
+    record.push("branch", Value::string(build::BRANCH, span));
 
-    let build_time = Some(build::BUILD_TIME).filter(|x| !x.is_empty());
-    if let Some(build_time) = build_time {
-        record.push("build_time", Value::string(build_time, call.head));
+    if let Some(commit_hash) = option_env!("NU_COMMIT_HASH") {
+        record.push("commit_hash", Value::string(commit_hash, span));
```

Comment:
Building all the infra and then slacking here :D

Reply:
Ha, that one's special :smile:. But actually, the macro returns `Option<String>`, so I can't reuse the function :(

Reply:
Ah right, didn't check the signature that closely